File: 263.Ugly Number.cpp
class Solution {
public:
    bool isUgly(int num) {
        if (num <= 0)
            return false;
        if (num == 1)
            return true;
            
        if (num % 2 == 0 || num % 3 == 0 || num % 5 == 0) {
            if (getDivideNum(num) == 1)
                return true;
        }
            
        return false;
    }
    
    int getDivideNum(int num) {
        if (num == 1)
            return num;
            
        if (num % 2 == 0)
            return getDivideNum(num/2);
        if (num % 3 == 0)
            return getDivideNum(num/3);
        if (num % 5 == 0)
            return getDivideNum(num/5);
            
        return num;
    }
};

// ================= improve ====================

for (int i=2; i<6 && num; i++)
    while (num % i == 0)
        num /= i;
return num == 1;

File: LeetCode/263.Ugly Number.py
class Solution(object):
    def isUgly(self, num):
        """
        :type num: int
        :rtype: bool
        """
        if num == 1:
            return True
        elif num <= 0:
            return False
        while num % 2 == 0:
            num >>= 1
        while num % 3 == 0:
            num /= 3
        while num % 5 == 0:
            num /= 5
        return num == 1
